### PR TITLE
pkg/tflite-micro: Update tflite-micro to latest version. requires update of pkg/flatbuffers as well. 

### PR DESCRIPTION
**Pull request by FlapKap**
### Contribution description
This PR updates tflite-micro to the the latest version. Since the current version in Riot OS is quite old, this has mandated some other changes and updates:
The included flatbuffers pkg and dist/tools have been updated to v23.5.26 as thats the one tflite-micro uses.
The patch files to fix cast-align errors have been updated
The mnist example have been updated with new locations for headers and i've replaced the AllOpsResolver with mutable_ops_resolver, since the former has been deprecated. This requires the user to specify the exact operators needed, with an added bonus of a much smaller binary size.

### Testing procedure
run the tests/pkg/tflite-micro test

### CODE REVIEW

**Comment by aabadie** [comment]
I tried to build locally and got this error:
```
"make" -C /work/riot/RIOT/build/pkg/tflite-micro/tensorflow/lite/micro -f /work/riot/RIOT/pkg/tflite-micro/tflite-micro.mk
In file included from /work/riot/RIOT/build/pkg/tflite-micro/tensorflow/lite/micro/micro_interpreter.cc:33:
/work/riot/RIOT/build/pkg/tflite-micro/tensorflow/lite/schema/schema_utils.h:29:38: error: 'OperatorCodeT' does not name a type; did you mean 'OperatorCode'?
   29 | BuiltinOperator GetBuiltinCode(const OperatorCodeT *op_code);
      |                                      ^~~~~~~~~~~~~
      |                                      OperatorCode
```

Any idea what it could be?

**Comment by aabadie** [comment]
Using `make clean all` fixes the problem.

**Comment by aabadie** [comment]
Can you prepend your 4 last commit messages with the right context ("pkg/tflite-micro: ", "pkg/flatbuffers: ") ?

**Comment by FlapKap** [comment]
> Can you prepend your 4 last commit messages with the right context ("pkg/tflite-micro: ", "pkg/flatbuffers: ") ?

done!

> I tried to build locally and got this error:
> 
> ```
> "make" -C /work/riot/RIOT/build/pkg/tflite-micro/tensorflow/lite/micro -f /work/riot/RIOT/pkg/tflite-micro/tflite-micro.mk
> In file included from /work/riot/RIOT/build/pkg/tflite-micro/tensorflow/lite/micro/micro_interpreter.cc:33:
> /work/riot/RIOT/build/pkg/tflite-micro/tensorflow/lite/schema/schema_utils.h:29:38: error: 'OperatorCodeT' does not name a type; did you mean 'OperatorCode'?
>    29 | BuiltinOperator GetBuiltinCode(const OperatorCodeT *op_code);
>       |                                      ^~~~~~~~~~~~~
>       |                                      OperatorCode
> ```
> 
> Any idea what it could be?

I encountered the same issue when i had generated the schema using a different version of flatc than required. Just using the schema_generated already in the repo fixed it.

**Comment by aabadie** [comment]
Can you fix the check-commits check? https://github.com/RIOT-OS/RIOT/actions/runs/9207542826/job/25327831907?pr=20683#step:4:5.
A multi line commit message should make it:
```
pkg/flatbuffers: downgrade to v23.5.26

because that's what tflite-micro depends on
```

**Comment by aabadie** [comment]
Sorry to bother you again with https://github.com/RIOT-OS/RIOT/pull/20683/commits/57abe1c406674bbf404fb8ac345c86078a649294
The commit message should be:
```
pkg/flatbuffers: downgrade to v23.5.26

because that's what tflite-micro depends on
```

format is: short description on the first line, blank line, extended description.

File: pkg/tflite-micro/Makefile [hunk 1]
```diff
@@ -40,5 +46,13 @@ all: $(TFLITE_MODULES_USED)
 
 .PHONY: tflite-%
 
-tflite-%:
+# .PHONY: $(PKG_SOURCE_DIR)/tensorflow/lite/schema/schema_generated.h
+
```
Comment (aabadie) [inline pkg/tflite-micro/Makefile:50]:
```suggestion
```

File: pkg/tflite-micro/patches/0001-fix-private-descriptor.patch [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+diff --git a/tensorflow/lite/micro/memory_planner/greedy_memory_planner.h b/tensorflow/lite/micro/memory_planner/greedy_memory_planner.h
+index b2cdb617..77b9a1cc 100644
+--- a/tensorflow/lite/micro/memory_planner/greedy_memory_planner.h
++++ b/tensorflow/lite/micro/memory_planner/greedy_memory_planner.h
+@@ -112,6 +112,8 @@ class GreedyMemoryPlanner : public MicroMemoryPlanner {
+   // that aren't being used during a phase of invocation are overwritten.
+   bool preserves_all_tensors() const override { return false; }
+ 
++  TF_LITE_REMOVE_VIRTUAL_DELETE
++
```
Comment (aabadie) [inline pkg/tflite-micro/patches/0001-fix-private-descriptor.patch:22]: This seems to add one or two trailing spaces

File: tests/pkg/tflite-micro/Makefile.ci [hunk 3]
```diff
@@ -1,109 +1,29 @@
 BOARD_INSUFFICIENT_MEMORY := \
-    airfy-beacon \
```
Comment (aabadie) [inline tests/pkg/tflite-micro/Makefile.ci:2]: This is removing too many boards to be true. I doubt all those removed boards have enough memory. nrf51dk has only 8kB of RAM.

Comment (FlapKap) [reply tests/pkg/tflite-micro/Makefile.ci:2]: I generated it through the `RIOT/dist/tools/insufficient_memory/create_makefile.ci.sh` script, but i must admit i don't have that board to test it on. Running `BOARD=airfy-beacon make -C tests/pkg/tflite-micro/` it builds fine, but I can't actually run it. The closest i have access to is a microbit, and this compiles and works on that

**Comment by FlapKap** [comment]
Murdock timed out on compiling the test for `arduino-mkrwan1300` after 300 seconds. It does compile on my machine, but it takes 6,5 minutes

**Pull request merged**